bolt/lib/Core/DebugData.cpp

```diff
@@ -177,6 +177,12 @@ uint64_t DebugRangesSectionWriter::getSectionOffset() {
   return SectionOffset;
 }
 
+void DebugRangesSectionWriter::appendToRangeBuffer(
+    const DebugBufferVector &CUBuffer) {
+  *RangesStream << CUBuffer;
+  SectionOffset = RangesBuffer->size();
```

Comment:
Are there cases where `SectionOffset` is out of sync with `RangesBuffer->size()`?

Reply:
Is always in sync, will be removed in a followup PR.

Reply:
Okay. Then you wouldn't need `appendToRangeBuffer()` as it becomes just `*RangesStream << CUBuffer`.

Reply:
We would still need `appendToRangeBuffer()` since `RangesStream` is a protected member and we call `appendToRangeBuffer()`  outside of `DebugRangesSectionWriter`, correct? But since it's a single line now we can define it in `DebugData.h` instead of `DebugData.cpp`.

Reply:
I guess it's fine.